src/ProxyManager/ProxyGenerator/LazyLoadingValueHolderGenerator.php

```diff
@@ -52,8 +53,10 @@ class LazyLoadingValueHolderGenerator implements ProxyGeneratorInterface
      *
      * @throws InvalidProxiedClassException
      * @throws InvalidArgumentException
+     *
+     * @psalm-param array{skipDestructor?: bool} $proxyOptions
      */
-    public function generate(ReflectionClass $originalClass, ClassGenerator $classGenerator)
+    public function generate(ReflectionClass $originalClass, ClassGenerator $classGenerator, array $proxyOptions = [])
```

Comment:
Because the class is not final, you may consider this as a BC break.
The solution I know about is to use `func_get_arg()` alongside an `@param` annotation that tells about the extra argument. If worth it, we could also trigger a runtime deprecation notice, that would be triggered when a child class doesn't declare the new argument.

Let me know how you'd like to deal with this.

Reply:
Using func_get_args in combination with a psalm-var annotation would make sense. IIRC this package does not announce deprecations via trigger_error, but you could do that in the lts package if so inclined.

Reply:
BC break now addressed using a few comments/annotations.